src/components/LawProposals.vue

```diff
@@ -12,34 +15,68 @@ export default defineComponent({
 
   data() {
     return {
+      laws: [] as Law[],
       selectedLaw: undefined as LawId | undefined,
+      fadeingLaw: undefined as LawId | undefined,
     }
   },
 
-  props: {
-    proposedLaws: Array as PropType<Law[]>,
+  mounted() {
+    const e = this.$el as HTMLElement
+    e.style.setProperty("--transitiontime", TRANSITION_TIME + "s")
+  },
+
+  computed: {
+    ...mapGetters(["proposedLaws"]),
   },
 
   methods: {
-    select(lawId: LawId) {
+    select(lawId: LawId | undefined) {
       this.selectedLaw = lawId
     },
 
     accept(lawId: LawId) {
-      this.store.dispatch("acceptLaw", { lawId })
+      this.fadeingLaw = lawId
+      setTimeout(() => this.store.dispatch("acceptLaw", { lawId }), TRANSITION_TIME * 1000)
     },
 
     decline(lawId: LawId) {
-      this.store.dispatch("rejectLaw", { lawId })
+      this.fadeingLaw = lawId
+      setTimeout(() => this.store.dispatch("rejectLaw", { lawId }), TRANSITION_TIME * 1000)
     },
   },
+
+  watch: {
+    proposedLaws(newLaws: Law[]) {
+      const laws: Law[] = [...this.laws]
+      const addedLaws: Law[] = newLaws.filter(nl => !laws.some(ol => ol.id === nl.id))
+
+      for (var i = 0; i < laws.length; i++) {
+        if (newLaws.some(nl => nl.id === laws[i].id)) {
+          continue
+        }
+        if (addedLaws.length === 0) {
+          laws.splice(i)
+          continue
+        }
+        laws[i] = addedLaws.shift() as Law
+      }
+      laws.push(...addedLaws)
+      this.laws = laws
+    }
```

Comment:
Es könnte praktisch sein, eine komplexe Funktion eher auszulagern (z.B. in den `LawProposer`) und hier nur noch aufzurufen. Das erleichtert das Unit-Testen.

Reply:
Hatte ich überlegt. Dann dachte ich, das hat ja mit Darstellung zutun und gehört deswegen nicht in die Algorithmen... aber Dein Unit-Testing-Argument überzeugt mich!

Reply:
Done. Allerdings noch ohne Unit-Test...